yt_dlp/extractor/generic.py

```diff
@@ -2814,6 +2829,18 @@ def _extract_embeds(self, url, webpage, *, urlh=None, info_dict={}):
                 # See https://github.com/ytdl-org/youtube-dl/pull/16735
                 pass
 
+        mobj = re.search(
+            r'<div [^>]*?class="jwplayer"[^>]*data-source=(?P<quote>[\'"])(?P<video_url>[^" >]+)(?P=quote)',
```

Comment:
```suggestion
            r'<div[^>]+\bclass=[\"\']jwplayer[\"\'][^>]+\bdata-source=([\"\'])(?P<video_url>(?:(?!\1).)+)\1',
```

slightly relaxed pattern that is generally used in other places too.

though this won't handle cases where the class and data-source are the other way round etc. there are functions in utils for better html extraction though they might not be stable enough for generic last time I checked.